lib/request/TingClientObjectRequest.php

```diff
@@ -148,12 +154,14 @@ public function processResponse(stdClass $response) {
     $objects = array();
     foreach ($response->collections as $collection) {
       foreach ($collection->objects as $object) {
-        $objects[$object->id] = $object;
+        $title = isset($object->record['dc:title'][''][0]) ? $object->record['dc:title'][''][0] : '';
+        // Ensure that getObject was able to finde the object.
+        if (strpos($title, self::MISSING_OBJECT_TITLE) !== 0) {
```

Comment:
This is very specific for the current state of search results.
Eventually openSearch will return an object->error element if the search fail for a given PID. 
Wouldn't it make more sense to add a object->error element if MISSING_OBJECT_TITLE is present in the title; and add some code to handle errors?
This way the code will be able to handle future versions of openSearch (or the current, for that matter, should errors on the object level be implemented. It's already in the XSD).

Reply:
Interesting suggestion :) This might be the way to go to ease the transition. The code that looks for the MISSING_OBJECT_TITLE would be obsolete, but I guess it doesn't hurt and can always be removed again.

Can we be sure that the error is present on the object after processing in `TingClientSearchRequest`? Tbh I'm not really sure about that when looking at the code.